### PR TITLE
mimx8mm GPIO

### PR DESCRIPTION
**Pull request by morandg**
This adds GPIO support for mimx8mm evk platform.

### CODE REVIEW

**Review by mnkp — DISMISSED** [review]

**Comment by morandg** [comment]
Just seen this PR https://github.com/zephyrproject-rtos/zephyr/pull/42428 and if gets merged, commit https://github.com/zephyrproject-rtos/zephyr/pull/42345/commits/14ae2f303aae5fdb9252ae0fed324194047e5ad2 could be consequently dropped!

**Review by mbolivar-nordic — DISMISSED** [review]

**Comment by carlescufi** [comment]
@dleach02 please approve and merge

**Comment by dleach02** [comment]
@morandg can you resolve the conflict

**Review by dleach02 — DISMISSED** [review]

**Comment by morandg** [comment]
> @morandg can you resolve the conflict

Done, thanks

**Review by pdgendt — DISMISSED** [review]

File: dts/arm/nxp/nxp_imx8m_m4.dtsi [hunk 1]
```diff
@@ -60,6 +61,76 @@
 			label = "OCRAM_S SYSTEM";
 		};
 
+		gpio1: gpio@30200000 {
+			compatible = "nxp,imx-gpio";
+			reg = <0x30200000 0x10000>;
```
[the remark this answers is not in the document]
Comment (morandg) [reply dts/arm/nxp/nxp_imx8m_m4.dtsi:66]: Yes, will fix

**Review by pdgendt — APPROVED** [review]

**Review by mnkp — APPROVED** [review]
LGTM, though I think this PR should be implemented in a single commit since all the changes are closely related.

**Comment by mbolivar-nordic** [comment]
@dleach02 please take another look

**Review by dleach02 — APPROVED** [review]

**Pull request merged**